go/runtime/scheduling/simple/txpool/ordered_map/ordered_map.go

```diff
@@ -0,0 +1,276 @@
+// Package ordered implements a tx pool backed by an ordered map.
+package ordered
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+	"github.com/oasisprotocol/oasis-core/go/runtime/scheduling/simple/txpool/api"
+)
+
+// Name is the name of the tx pool implementation.
+const Name = "ordered-map"
+
+var _ api.TxPool = (*orderedMap)(nil)
+
+type pair struct {
+	Key   hash.Hash
+	Value []byte
+
+	element *list.Element
+}
+
+type orderedMap struct {
+	sync.Mutex
+
+	calls map[hash.Hash]*pair
```

Comment:
Since we're refactoring all of this, we can rename calls to transactions everywhere (fields, method names, ...) to make it more clear.

Reply:
:+1:  Done, and also all other comments.